README.md

```diff
@@ -18,52 +18,106 @@ The NTT manager contracts additionally keep track of the token decimals of the o
 
-3. Transmit
+3. **Transmit**
 
 Once the NttManager forwards the message to the Transceiver the message is transmitted via the [`sendMessage`] method. The method signature if enforced by the [`Transceiver`] but transceivers are free to determine their own implementation for transmitting messages.
 (e.g A message routed through the Wormhole Transceiver can be sent via automatic relaying (AR), via a specialized or custom relayer, or via the core bridge).The following event is emitted once the message has been transmitted.
 ``` solidity
-emit SendTransceiverMessage(recipientChain, endpointMessage);
+    event SendTransceiverMessage(
+        uint16 recipientChain, TransceiverStructs.TransceiverMessage message
+    );
 ```
-4. Receive
+4. **Receive**
 
 Once a message has been transmitted across the wire, an off-chain process (e.g. a relayer) will forward the message to the corresponding Transceiver on the recipient chain. The relayer interacts with the transceiver via an entrypoint for receiving messages (e.g. Wormhole messages are received through the [`receiveWormholeMessages`] method, which performs the messages verification along with replay protection)
 The following event is emitted during this process:
 ``` solidity
-emit ReceivedRelayedMessage(deliveryHash, sourceChain, sourceAddress);
+    event ReceivedRelayedMessage(bytes32 digest, uint16 emitterChainId, bytes32 emitterAddress);
 ```
 This method should also forward the message to the NttManager on the recipient chain.
 NOTE: The Transceiver interface does not enforce the method signature abstractly because receiving messages may be specific to the way in which a transceiver consumes messages.
 
+    a. Messages received on the destination chains are also rate limited as described in (link to the section on Rate-Limiting and Cancel Flows). If an inbound transfers is rate-limited, the following event is emitted:
+
 ``` solidity
-emit ReceivedMessage(vm.hash, vm.emitterChainId, vm.emitterAddress, vm.sequence);
+event InboundTransferQueued(bytes32 digest);
 ```
 
-5. Attestation
+If a user attempts to release the transfers before the rate-limit duration, we revert with the following error messages:
 
 ``` solidity
-emit MessageAttestedTo(digest, endpoint, _getEndpointInfosStorage()[endpoint].index);
-emit MessageAlreadyExecuted(sourceManagerAddress, digest);
+error InboundQueuedTransferStillQueued(bytes32 digest, uint256 transferTimestamp);
 ```
 
-6. Mint or Unlock
+5. **Attest**
+
+Once the message is delivered to the Ntt Manager, this contract updates its current state to reflect the number of attestations
+the message has received from distinct transceivers.
+On each attesation, this event is emitted:
+
 ``` solidity
-emit TransferRedeemed(digest);
+event MessageAttestedTo(bytes32 digest, address transceiver, uint8 index);
+```
+
+Based on these attestations, the contract decides whether the can be executed.
```

Comment:
The contract uses a threshold requirement to determine whether a message can be executed after attestations are received. For example, if the threshold requirement is `1`, the message will be executed after a single Transceiver delivers an attestation. If the threshold requirement is `2`, the message will only be executed after two Transceivers deliver attestations.